integration/integration_test.go

```diff
@@ -7891,6 +7893,325 @@ func getRemoteAddrString(sshClientString string) string {
+	_, err = aSrv.CreateRole(ctx, sshAccessRole)
+	require.NoError(t, err)
+
+	peerRole, err := types.NewRole("peer", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			RequireSessionJoin: []*types.SessionRequirePolicy{
+				{
+					Name:   "Requires oversight",
+					Filter: `equals("true", "true")`,
+					Kinds: []string{
+						string(types.SSHSessionKind),
+					},
+					Count: 1,
+					Modes: []string{
+						string(types.SessionModeratorMode),
+					},
+					OnLeave: string(types.OnSessionLeaveTerminate),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	_, err = aSrv.CreateRole(ctx, peerRole)
+	require.NoError(t, err)
+
+	peerUser, err := types.NewUser(peerUsername)
+	require.NoError(t, err)
+	peerUser.SetLogins([]string{username})
+	peerUser.SetRoles([]string{sshAccessRole.GetName(), peerRole.GetName()})
+	_, err = aSrv.CreateUser(ctx, peerUser)
+	require.NoError(t, err)
+
+	modUsername := username + "-moderator"
+	modRole, err := types.NewRole("moderator", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			JoinSessions: []*types.SessionJoinPolicy{{
+				Name:  "Session moderator",
+				Roles: []string{peerRole.GetName()},
+				Kinds: []string{string(types.SSHSessionKind)},
+				Modes: []string{string(types.SessionModeratorMode), string(types.SessionObserverMode)},
+			}},
+		},
+	})
+	require.NoError(t, err)
+	_, err = aSrv.CreateRole(ctx, modRole)
+	require.NoError(t, err)
+
+	modUser, err := types.NewUser(modUsername)
+	require.NoError(t, err)
+	modUser.SetLogins([]string{username})
+	modUser.SetRoles([]string{sshAccessRole.GetName(), modRole.GetName()})
+	_, err = aSrv.CreateUser(ctx, modUser)
+	require.NoError(t, err)
+
+	waitForNodesToRegister(t, instance, helpers.Site)
+
+	// Start a shell so a moderated session is created
+	peerTC, err := instance.NewClient(helpers.ClientConfig{
+		TeleportUser: peerUsername,
+		Login:        username,
+		Cluster:      helpers.Site,
+		Host:         Host,
+	})
+	require.NoError(t, err)
+
+	peerClusterCli, err := peerTC.ConnectToCluster(ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_ = peerClusterCli.Close()
+	})
+
+	nodeDetails := client.NodeDetails{
+		Addr:      instance.Config.SSH.Addr.Addr,
+		Namespace: peerTC.Namespace,
+		Cluster:   helpers.Site,
+	}
+	peerNodeCli, err := peerTC.ConnectToNode(
+		ctx,
+		peerClusterCli,
+		nodeDetails,
+		username,
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, peerNodeCli.Close())
+	})
+
+	peerSSH := peerNodeCli.Client
+	cmdSess, err := peerSSH.NewSession(ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, cmdSess.Close())
+	})
+
+	peerTerm := NewTerminal(250)
+	cmdSess.Stdin = peerTerm
+	cmdSess.Stdout = peerTerm
+	cmdSess.Stderr = peerTerm
+	err = cmdSess.Shell(ctx)
+	require.NoError(t, err)
+
+	var sessTracker types.SessionTracker
+	require.Eventually(t, func() bool {
+		trackers, err := peerClusterCli.AuthClient.GetActiveSessionTrackers(ctx)
+		require.NoError(t, err)
```

Comment:
This will fail the entire test even if the check fails on the first iteration, when the behavior you want is that it can fail for up to 5 seconds before the test should fail.

Recommend using `require.EventuallyWithT` here.

Reply:
Done